### PR TITLE
Fix dashboard tests

### PR DESCRIPTION
**Pull request by svtkachenko**
Test results:
```
py.test -v --junit-xml=foreman-results.xml -m 'not stubbed' tests/foreman/ui/test_dashboard.py
============================= test session starts ==============================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/jenkins/shiningpanda/jobs/ad0ef6b5/virtualenvs/d41d8cd9/bin/python2.7
cachedir: .cache
rootdir: /home/jenkins/workspace/satellite6-standalone-automation, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0
collecting ... collected 25 items

tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_add_removed_widget PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_clear_search_box PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_content_host_subscription_status <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_content_view_history PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_current_subscription_totals PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_host_collections PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_host_configuration_chart PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_host_configuration_status PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_latest_warning_error_tasks <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_remove_widget PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_rendering_after_env_removed PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_reset PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_save PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_search PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_search_random PASSED
tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_sync_overview_widget PASSED

tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_task_status PASSED

tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_toggle_auto_refresh PASSED

tests/foreman/ui/test_dashboard.py::DashboardTestCase::test_positive_user_access_with_host_filter <- robottelo/decorators/__init__.py PASSED

============================== 6 tests deselected ==============================
============ 18 passed, 1 skipped, 6 deselected in 3260.74 seconds =============
```

### CODE REVIEW

File: tests/foreman/ui/test_dashboard.py [hunk 1]
```diff
@@ -233,58 +233,12 @@ def test_positive_add_removed_widget(self):
...
-            for widget in ['Discovered Hosts', 'Content View History']:
+            for widget in ['Discovered Hosts', 'Content Views']:
                 self.dashboard.remove_widget(widget)
                 self.dashboard.manage_widget('Save Dashboard')
                 self.dashboard.manage_widget('Add Widget', widget)
                 self.assertIsNotNone(self.dashboard.get_widget(widget))
 
-    @run_in_one_thread
-    @tier1
-    def test_positive_minimize_widget(self):
```
Comment (svtkachenko) [inline tests/foreman/ui/test_dashboard.py]: Minimize functionality was removed. See https://github.com/theforeman/foreman/pull/4351

File: tests/foreman/ui/test_dashboard.py [hunk 2]
```diff
@@ -442,16 +396,17 @@ def test_positive_task_status(self):
         with Session(self.browser) as session:
             set_context(session, org=org.name)
             self.assertTrue(self.dashboard.validate_task_navigation(
-                'pending', 'state=running&result=pending'))
+                'running', 'state=running&result=pending'))
```
Comment (svtkachenko) [inline tests/foreman/ui/test_dashboard.py]: Not only running tasks have status `pending`.

File: tests/foreman/ui/test_dashboard.py [hunk 3]
```diff
@@ -442,16 +396,17 @@ def test_positive_task_status(self):
...
             self.assertTrue(self.dashboard.validate_task_navigation(
                 'success',
                 'state=stopped&result=success',
                 "Publish content view '{0}'; organization '{1}'".format(
                     content_view.name, org.name)
             ))
             self.assertTrue(self.dashboard.validate_task_navigation(
-                'error', 'state=stopped&result=error'))
+                'warning', 'state=stopped&result=warning'))
```
Comment (svtkachenko) [inline tests/foreman/ui/test_dashboard.py]: I wasn't able to get tasks 'error' with `error` status on clean machine, but 'warning` was present. Probably it caused by BZ [1] and failed tasks do not obtain `error` status.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1460240

File: robottelo/ui/dashboard.py [hunk 4]
```diff
@@ -10,21 +10,6 @@
 class Dashboard(Base):
     """Manipulates dashboard from UI"""
 
-    widgets_locators = {
```
Comment (svtkachenko) [inline robottelo/ui/dashboard.py]: Now `@data-name` has same value as displayed widget name, so this workaround is not needed.

File: robottelo/ui/dashboard.py [hunk 5]
```diff
@@ -65,29 +50,22 @@ def remove_widget(self, widget_name):
...
             wait_for_ajax=False
         )
         self.handle_alert(True)
         self.wait_until_element_is_not_visible(
-            locators['dashboard.remove_widget'] %
-            self.widgets_locators[widget_name]
-        )
+            locators['dashboard.remove_widget'] % widget_name)
 
     def minimize_widget(self, widget_name):
```
Comment (oshtaier) [inline robottelo/ui/dashboard.py]: Shouldn't it removed entirely?

Comment (svtkachenko) [inline robottelo/ui/dashboard.py]: Upd: removed.

**Comment by oshtaier** [comment]
ACK

**Pull request merged**